src/app/code/community/Shopgate/Cloudapi/Model/Observers/AddSaleRuleTypeToQuote.php

```diff
@@ -31,14 +31,12 @@ public function execute(Varien_Event_Observer $observer)
     {
         /** @var Mage_Sales_Model_Quote $quote */
         $quote = $observer->getData('quote');
-        if ($quote->hasData(Shopgate_Cloudapi_Model_SalesRule_Condition::CART_TYPE)
-            || !Mage::helper('shopgate_cloudapi/request')->isShopgateApi()
-        ) {
+        if ($quote->getData(Shopgate_Cloudapi_Model_SalesRule_Condition::CART_TYPE) !== null) {
```

Comment:
still needs to check if it's a shopgateApi call, we don't want to run this logic on regular calls outside our REST. Does this also need to run in Framework? Oo

Reply:
If I would add a check for shopgateApi , this will not be triggered in the web checkout, which invalidates app only coupons?!